src/riscv/lib/src/jit/state_access.rs

```diff
@@ -308,6 +312,32 @@ extern "C" fn reservation_set_read<MC: MemoryConfig, M: ManagerReadWrite>(
     <MachineCoreState<MC, M> as ICB>::reservation_set_read(core)
 }
 
+/// Convert an unsigned 64-bit `XValue` to a 64-bit `FValue` using the given rounding mode.
+extern "C" fn f64_from_x64_unsigned<MC: MemoryConfig, M: ManagerReadWrite>(
+    core: &mut MachineCoreState<MC, M>,
+    exception_out: &mut MaybeUninit<Exception>,
+    xval: XValue,
+    rm: u32,
+    fval_out: &mut MaybeUninit<FValue>,
+) -> bool {
+    if let Some(round_mode) = InstrRoundingMode::from_rm(rm) {
```

Comment:
There is no good reason to perform this check at runtime. The rounding mode is known at JIT compilation time.